### PR TITLE
Adopt a code of conduct for public repositories

### PR DESCRIPTION
**Pull request by lawrence-forooghian**
This comes from a conversation at the diversity and inclusion group.

https://trello.com/c/7NR54D1Z/78-adopt-a-code-of-conduct-by-default-for-our-code

### CODE REVIEW

File: rfc-028-adopt-a-code-of-conduct.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+about whether the project is seeking contributions, and whether we’re likely to
+accept them, then we’re fine. We might like to reply to people’s contributions
+and explain why we’re not going to accept them, even if it takes up a little of
+our time.
+
+## Next steps
+
+- Set up an email contact code-of-conduct@dxw.com that goes to Dominic Baggott
+- Set up an email contact code-of-conduct-md@dxw.com that goes to David Mann
+- Confirm that Dave and Dom are happy to take this role
```
Comment (lawrence-forooghian) [inline rfc-028-adopt-a-code-of-conduct.md:116]: @evilstreak (and does Dave have a GitHub account) – I'd like to hear your thoughts on being the reporting / enforcement folks, when you've got a moment to think about this.

File: rfc-028-adopt-a-code-of-conduct.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+The `CONTRIBUTING.md` file:
+
+- SHOULD explain whether the project is actively seeking contributions from
+  outside dxw
+- MUST link to the `CODE_OF_CONDUCT.md` file
+
+The `CODE_OF_CONDUCT.md` file:
+
+- SHOULD be a copy of the [Markdown version of the Contributor Covenant](https://www.contributor-covenant.org/version/2/0/code_of_conduct/).
```
Comment (erbridge) [inline rfc-028-adopt-a-code-of-conduct.md:57]: There's a CLI to automate some of this, if you think that's relevant: https://github.com/simonv3/covenant-generator

File: rfc-028-adopt-a-code-of-conduct.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+- SHOULD be modified to include the following information about contact methods:
+  > Instances of abusive, harassing, or otherwise unacceptable behavior may be
+  > reported to dxw’s Chief Technology Officer at code-of-conduct@dxw.com. If a
+  > complaint relates to the conduct of the Chief Technology Officer, then
+  > complaints may be reported to the Managing Director at
+  > code-of-conduct-md@dxw.com. The current holders of these roles can be found
+  > on [dxw’s people page](https://www.dxw.com/our-people/).
```
Comment (erbridge) [inline rfc-028-adopt-a-code-of-conduct.md:64]: 👍

The fallback is good.

File: rfc-028-adopt-a-code-of-conduct.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+  outside dxw
+- MUST link to the `CODE_OF_CONDUCT.md` file
+
+The `CODE_OF_CONDUCT.md` file:
+
+- SHOULD be a copy of the [Markdown version of the Contributor Covenant](https://www.contributor-covenant.org/version/2/0/code_of_conduct/).
+- SHOULD be modified to include the following information about contact methods:
+  > Instances of abusive, harassing, or otherwise unacceptable behavior may be
+  > reported to dxw’s Chief Technology Officer at code-of-conduct@dxw.com. If a
+  > complaint relates to the conduct of the Chief Technology Officer, then
```
Comment (erbridge) [inline rfc-028-adopt-a-code-of-conduct.md:61]: I think Dom is "Director of Technology" (to be pedantic).

File: rfc-028-adopt-a-code-of-conduct.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+When starting a client project, we SHOULD speak to the client to find out
+whether they would like us to add a code of conduct to the project. We’ll need
+to understand whether they:
+
+- have the ability to accept reports of violations
+- can do this privately
+- want to take on the responsibility of enforcing the code of conduct
+
+dxw SHOULD NOT take on the responsibility of enforcing the Code of Conduct for
+a client project.
```
Comment (erbridge) [inline rfc-028-adopt-a-code-of-conduct.md:75]: 👍

File: rfc-028-adopt-a-code-of-conduct.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+TODO Something about enforcement? – this is one to check with Dom and Dave too.
+This is from the Contributor Covenant site:
```
Comment (erbridge) [inline rfc-028-adopt-a-code-of-conduct.md:78]:
```suggestion
The Director of Technology or Managing Director MUST take action when receiving a violation report. At a minimum, this MUST include reposponding to the reporter with a plan of action. We MUST document and SHOULD publicise our enforcement procedures as they develop.
```

I think we can be iterative about it.

File: rfc-028-adopt-a-code-of-conduct.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+>
+> Before adopting the Contributor Covenant take the time to discuss and
+> decide how to deal with problems as they emerge. Document the policy and
+> procedure for enforcement, and add it to your README or in another visible,
+> appropriate place. Consider if your project team has the willingness and
+> maturity to follow through on your enforcement procedures.
+
+## Concerns
+
+Adding a `CONTRIBUTING.md` might suggest that we’re actively seeking
```
Comment (erbridge) [inline rfc-028-adopt-a-code-of-conduct.md:97]:
```suggestion
Adding a code of conduct might suggest that we’re actively seeking
```

File: rfc-028-adopt-a-code-of-conduct.md (hunk 2 above)
Comment (lawrence-forooghian) [reply rfc-028-adopt-a-code-of-conduct.md:57]: That's an option, although if we do the "Create fallback CONTRIBUTING.md and CODE_OF_CONDUCT.md files for the dxw GitHub organisation" step then we’ll just be able to copy it from dxw's `.github` repository, might be easier.

File: rfc-028-adopt-a-code-of-conduct.md (hunk 4 above)
Comment (lawrence-forooghian) [reply rfc-028-adopt-a-code-of-conduct.md:61]: On dxw.com/our-people he’s "Chief Technology Officer"?

File: rfc-028-adopt-a-code-of-conduct.md (hunk 6 above)
Comment (lawrence-forooghian) [reply rfc-028-adopt-a-code-of-conduct.md]: 👍 – added your suggested words, thanks.

File: rfc-028-adopt-a-code-of-conduct.md (hunk 7 above)
Comment (lawrence-forooghian) [reply rfc-028-adopt-a-code-of-conduct.md:84]: I think the contributing doc is the one which explains how people can contribute to the project, and so it’s the one which would imply we’re looking for contributions?

Comment (erbridge) [reply rfc-028-adopt-a-code-of-conduct.md:84]: What we're sugesting is addnig a `CONTRIBUTING.md` that explicitly states that we're not accepting contributions, though, aren't we?

Comment (lawrence-forooghian) [reply rfc-028-adopt-a-code-of-conduct.md]: True – I've changed it.

File: rfc-028-adopt-a-code-of-conduct.md (hunk 1 above)
Comment (evilstreak) [reply rfc-028-adopt-a-code-of-conduct.md:103]: I've chatted this through with Dave. We're both happy to take on these roles.

**Pull request merged**